proto/viam/app/build/v1/build.proto

```diff
@@ -0,0 +1,82 @@
+syntax = "proto3";
+
+package viam.app.build.v1;
+
+import "google/protobuf/timestamp.proto";
+
+option go_package = "go.viam.com/api/app/build/v1";
+
+// Note on ID types:
+// build_id: The 'parent' ID for a set of builds of module@version. Builds contain multiple jobs. This ID is globally unique.
+
+// 2023-11-28: Unstable API. Will change.
+service BuildService {
+  // Asynchronously start a build of a Viam module, with one job per platform.
+  rpc StartBuild(StartBuildRequest) returns (StartBuildResponse);
+  // Fetch logs from a single job of a single build.
+  rpc GetLogs(GetLogsRequest) returns (stream GetLogsResponse);
+  // List the jobs for a module ordered by (build start time, alphabetical platform).
+  rpc ListJobs(ListJobsRequest) returns (ListJobsResponse);
+}
+
+message StartBuildRequest {
+  // repo to build
+  string repo = 1;
+  // optional git ref; defaults to 'main'
+  optional string ref = 2;
+  // list of platforms to build
+  repeated string platform = 3;
+  // module_id as prefix:name pair
+  string module_id = 4;
+}
+
+message StartBuildResponse {
+  string build_id = 1;
+}
+
+message GetLogsRequest {
+  string build_id = 1;
+  string platform = 2;
+}
+
+// GetLogsResponse is a streaming endpoint that may have multiple messages that belong
+// to the same build_step if there are too many bytes to fit into a single gRPC
+// response.
+message GetLogsResponse {
```

Comment:
This might need to be a streaming API. How long are the logs? I forget our max length for a GRPC request, but given that we don't know the length and could get really long... Its probably best to just make it streaming now and avoid issues later.

Reply:
getlogs api returns a stream

Reply:
It is a streaming API, that is what the comment says. Did I misunderstand your comment?

Reply:
Good question. The reason I wrote it like this is primarily laziness (the build step will be ~10 bytes and the data string will be ~4MB so repeating the build step  a few times didn't really seem like a big deal). However, if this api is also confusing, that is bad. I will fix this

Reply:
> Does it work without oneof?

Yes, it just wastes a few bytes per message that shares the same build step

Reply:
Ok, I learned it isn't required today. I see other APIs that don't do it now too. I feel like the "convention" should be to use `oneof` but I think there are more usages without it than with. So we can pass on this.